platforms/dragonboard/dragonboard_adaptor.go

```diff
@@ -0,0 +1,193 @@
+package dragonboard
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"gobot.io/x/gobot"
+	"gobot.io/x/gobot/drivers/i2c"
+	"gobot.io/x/gobot/sysfs"
+)
+
+// Adaptor represents a Gobot Adaptor for a C.H.I.P.
+type Adaptor struct {
+	name        string
+	digitalPins map[int]sysfs.DigitalPin
+	pinMap      map[string]int
+	i2cBuses    [3]sysfs.I2cDevice
+}
+
+var fixedPins = map[string]int{
+	"GPIO_A": 36,
+	"GPIO_B": 12,
+	"GPIO_C": 13,
+	"GPIO_D": 69,
+	"GPIO_E": 115,
+	"GPIO_F": 507,
+	"GPIO_G": 24,
+	"GPIO_H": 25,
+	"GPIO_I": 35,
+	"GPIO_J": 34,
+	"GPIO_K": 28,
+	"GPIO_L": 33,
+
+	"LED_1": 21,
+	"LED_2": 120,
+}
+
+// NewAdaptor creates a C.H.I.P. Adaptor
+func NewAdaptor() *Adaptor {
+	c := &Adaptor{
+		name:        gobot.DefaultName("DragonBoard"),
+		digitalPins: make(map[int]sysfs.DigitalPin),
+	}
+
+	c.setPins()
+	return c
+}
+
+// Name returns the name of the Adaptor
+func (c *Adaptor) Name() string { return c.name }
+
+// SetName sets the name of the Adaptor
+func (c *Adaptor) SetName(n string) { c.name = n }
+
+// Connect initializes the board
+func (c *Adaptor) Connect() (err error) {
+	return
+}
+
+// Finalize closes connection to board and pins
+func (c *Adaptor) Finalize() (err error) {
+	for _, pin := range c.digitalPins {
+		if pin != nil {
+			if e := pin.Unexport(); e != nil {
+				err = multierror.Append(err, e)
+			}
+		}
+	}
+	for _, bus := range c.i2cBuses {
+		if bus != nil {
+			if e := bus.Close(); e != nil {
+				err = multierror.Append(err, e)
+			}
+		}
+	}
+	return
+}
+
+func (c *Adaptor) setPins() {
+	c.pinMap = fixedPins
+	baseAddr, _ := getXIOBase()
```

Comment:
This currently just maps the entire gpio device, actual use case would be to use GPIO_A..L or UART0_TxD, not the GPIO_0..122, This will be removed once all the IO has been mapped and drivers for most peripherals are working.